src/executor/evaluate/mod.rs

```diff
@@ -281,6 +283,72 @@ async fn evaluate_function<'a, T: 'static + Debug>(
 
             Ok(Evaluated::from(Value::Str(converted)))
         }
+        Function::ASin(expr) => {
+            let number = match eval(expr).await?.try_into()? {
+                Value::F64(v) => Some(v),
+                Value::Str(v) => match f64::from_str(&v) {
+                    Ok(f) => Some(f),
+                    Err(_) => None,
+                },
+                Value::I64(v) => f64::from_i64(v),
+                _ => None,
```

Comment:
Now, let's make the spec clear.

You can only accept three types; `F64`, `I64` and `Null`.
Except for these three types, `FunctionReuqiresFloatValue` can be the one to return.
`F64` - we can use directly.
`I64` - `as f64` to cast and we can use that.
`Null` -> then return value becomes also null.